src/foam/core/types.js

```diff
@@ -98,6 +98,20 @@ foam.CLASS({
       value: function (_, d) {
         if ( typeof d === 'number' ) return new Date(d);
         if ( typeof d === 'string' ) {
+
+          // Currently we are allowing users to input >4-digit years 
+          // in our DateViews, but the ISO string is not formatted correctly
+          // A valid ISO string for >4-digit year must be padded to 6 digits
+          // in the year field and prepended by '+'
+          // this fixes the problem of the invalid string causing error to be
+          // thrown below.
+          year = d.substring(0, d.indexOf('-'));
```

Comment:
We shouldn't need to do any of this. I was looking at this with Anna earlier and the server is sending us an invalid date so the fix should be there and not here. `src/foam/lib/json/Outputter.java` uses a hand crafted date formatter that attempts to do ISO format but it's incorrect. It should use a canned formatter that java already provides. One of these: https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html